src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -305,8 +305,10 @@ public ConsumerDTO createConsumerInEnvironment(String envId, ConsumerDTO consume
         this.validator.validateCollectionElementsNotNull(consumer::getInstalledProducts,
             consumer::getGuestIds, consumer::getCapabilities);
 
+        //TODO: This needs to support backward compatibility.
+        // Kept as it is to be handled in other task of multi-environment feature.
         Environment e = lookupEnvironment(envId);
-        consumer.setEnvironment(translator.translate(e, EnvironmentDTO.class));
+        consumer.addEnvironments(translator.translate(e, EnvironmentDTO.class));
```

Comment:
If the consumer DTO provided as part of this request already includes environments, we would add the environment requested explicitly as part of the resource to the end of that list and then do a normal creation, rather than creating the consumer in the single requested environment.

Is that expected behavior here?

Reply:
The clients calling this endpoint (legacy subman clients) have no  (and will not ever have) knowledge that you can set environments in the request body; but just to be sure, perhaps we should do a `.clear()` before adding the env from the URL param

Reply:
Oh, we still need to handle this. Probably by way of either .clear as nikos suggested, or by way of using .setEnvironments (which is my preferred method).

Reply:
Updated to use `.setEnvironments`. I hope this will be done in a more detailed manner in the next tasks.